File: gc-starter/crud-spring-boot-starter/src/main/java/com/gc/starter/crud/model/Sort.java
package com.gc.starter.crud.model;

import lombok.Getter;
import org.springframework.lang.NonNull;

/**
 * 排序字段
 * @author shizhongming
 * 2020/1/12 4:13 下午
 */
@Getter
public final class Sort {
    @NonNull
    private String name;

    @NonNull
    private String order;

    /**
     * 对应的数据库字段
     */
    @NonNull
    private String dbName;

    public Sort(@NonNull String name, @NonNull String order, @NonNull String dbName) {
        this.name = name;
        this.order = order;
        this.dbName = dbName;
    }
}

File: smart-starter/crud-spring-boot-starter/src/main/kotlin/com/smart/starter/crud/model/Sort.kt
package com.smart.starter.crud.model

/**
 * 排序字段
 * @author ming
 * 2019/6/12 上午10:42
 */
class Sort(val name: String, val order: String = "asc") {

    // 数据库名称
    lateinit var dbName: String
}